fe1-web/src/core/objects/Server.ts

```diff
@@ -0,0 +1,63 @@
+import { RemoveMethods } from 'core/types';
+
+import { PublicKey } from './PublicKey';
+
+export type ServerAddress = string;
+
+export interface ServerState {
+  address: string;
+  publicKey: string;
+}
+
+export class Server {
+  /**
+   * The canonical address of the server
+   */
+  address: ServerAddress;
+
+  /**
+   * The public key of the server that can be used to send encrypted messages
+   */
+  publicKey: PublicKey;
+
+  // NOTE: There is no need to store peers: ServerAddress[] here.
+  // As soon as a greeting message arrives, we connect to all peers. The server addresses
+  // will be added to the lao state as soon as a lao creation message is received
+  // over each connection
+
+  /**
+   * Constructs a new server instance
+   * @param server The properties of the new server instance
+   */
+  constructor(server: RemoveMethods<Server>) {
```

Comment:
Outside the scope of this PR:
I can't help but feel that `RemoveMethods` is the wrong name when reading this.
It is a type, but the name implies a verb (describing how it's implemented).
If we focus on its usage, how about something like `AttributesOnly<>` or `DataFields<>` ?

Reply:
Yeah I agree!

Reply:
I like `AttributesOnly` or `OmitMethods` (analogous to `Omit`) would work too